python/ray/tune/sync_client.py

```diff
@@ -418,3 +433,235 @@ def _validate_exclude_template(exclude_template):
                     "Neither `{pattern}` nor `{regex_pattern}` found in "
                     f"exclude string `{exclude_template}`"
                 )
+
+
+def _get_recursive_files_and_stats(path: str) -> Dict[str, Tuple[float, int]]:
+    """Return dict of files mapping to stats in ``path``.
+
+    This function scans a directory ``path`` and returns a dict mapping
+    each contained file to a tuple of (mtime, filesize).
+    """
+    files_stats = {}
+    for root, dirs, files in os.walk(path, topdown=False):
+        rel_root = os.path.relpath(root, path)
+        for file in files:
+            key = os.path.join(rel_root, file)
+            stat = os.lstat(os.path.join(path, key))
+            files_stats[key] = round(stat.st_mtime, 5), stat.st_size
```

Comment:
Should be robust enough for now (it's the same that aws uses). We can add an option for hashes later but it will be large overhead to do this everytime. Thus we would need to cache this - let's keep it like this for now and see if we run into problems.

Reply:
any thoughts behind round(st_mtime, 5)? add it as a comment?

Reply:
Actually I think we can remove this